android/src/main/java/com/reactnativecommunity/imageeditor/ImageEditorModule.java

```diff
@@ -467,23 +472,27 @@ private static void writeCompressedBitmapToFile(Bitmap cropped, String mimeType,
    *
    * @param mimeType the MIME type of the file to create (image/*)
    */
-  private static File createTempFile(Context context, @Nullable String mimeType)
+  private static File createTempFile(Context context, @Nullable String mimeType, boolean useInternalCache)
       throws IOException {
     File externalCacheDir = context.getExternalCacheDir();
     File internalCacheDir = context.getCacheDir();
-    File cacheDir;
+    File cacheDir = null;
     if (externalCacheDir == null && internalCacheDir == null) {
       throw new IOException("No cache directory available");
     }
-    if (externalCacheDir == null) {
+
+    if (useInternalCache || externalCacheDir == null) {
       cacheDir = internalCacheDir;
+    } else if (externalCacheDir != null && internalCacheDir != null) {
+       cacheDir = (externalCacheDir.getFreeSpace() > internalCacheDir.getFreeSpace() ?
+          externalCacheDir : internalCacheDir );
     }
-    else if (internalCacheDir == null) {
-      cacheDir = externalCacheDir;
-    } else {
-      cacheDir = externalCacheDir.getFreeSpace() > internalCacheDir.getFreeSpace() ?
-          externalCacheDir : internalCacheDir;
+
+    if (cacheDir == null) {
+      throw new IOException("Could not access a cache directory");
     }
```

Comment:
Maybe add this case above in 
```
    if (externalCacheDir == null && internalCacheDir == null) {
      throw new IOException("No cache directory available");
    }
```

You would need 

```
    if ((externalCacheDir == null && internalCacheDir == null) || (internalCacheDir == null && useInternalCache)) {
      throw new IOException("No cache directory available");
    }
```

Or throw the same error at least - could be some people depend on the error message

Reply:
Fixed :)